lib/server.js

```diff
@@ -150,33 +161,36 @@ var routes = [
     }
   },
 
-  {
+  { // slack retrigger command
     url: /^\/slack.*/,
     handler: function(request, respond, onError) {
-      var patchId = request.text;
-      var fragments = decodeURIComponent(request.url).
-        match(/[\/?&]text=([^&]+)/)[1].
-        split(" ");
-
-      var jobName = fragments[0];
-      var buildId = fragments[1];
-
-      var jobLink = [ K.JENKINS_URL, 'job', jobName, buildId ].join("/");
+      var jobLink = slackUrlToJenkins(request.url)
 
       jenkins.connect().then(function() {
         return jenkins.retrigger(jobLink);
       }).then(function() {
         return jenkins.getJob(jobLink);
-      }).then(function(jobStatus) {
-        respond(200, jobStatus);
+      }).then(function() {
+        respond(200, "Your job has been retriggered.");
+      }, onError);
+    }
+  },
+
+  { // slack job status command
+    url: /^\/jobstatus.*/,
+    handler: function(request, respond, onError) {
+      var jobLink = slackUrlToJenkins(request.url)
+
+      jenkins.getJob(jobLink).then(function(jobStatus) {
+        respond(200, "Current job status: " + jobStatus.status);
       }, onError);
     }
   },
 
   { // 404
-    url: /.*/,
-    handler: function(request, respond) {
-      respond(404, { message: K.ERROR_NOT_FOUND });
+    url: /^\/incoming.*/,
```

Comment:
What is `incoming`? the diff shows that the 404 handler is gone, is that so? why?